pkg/statistics/fmsketch.go

```diff
@@ -46,18 +46,29 @@ var fmSketchPool = sync.Pool{
 // TODO: add this attribute to PB and persist it instead of using a fixed number(executor.maxSketchSize)
 const MaxSketchSize = 10000
 
-// FMSketch (Flajolet–Martin Sketch) is a probabilistic data structure used for estimating the number of distinct elements in a stream.
-// It uses a hash function to map each element to a binary number and counts the number of trailing zeroes in each hashed value.
-// The maximum number of trailing zeroes observed gives an estimate of the logarithm of the number of distinct elements.
-// This approach allows the FM sketch to handle large streams of data in a memory-efficient way.
-//
-// See https://en.wikipedia.org/wiki/Flajolet%E2%80%93Martin_algorithm
+// FMSketch (Flajolet–Martin Sketch) is a probabilistic data structure that estimates the count of unique elements in a stream.
+// It employs a hash function to convert each element into a binary number and then counts the trailing zeroes in each hashed value.
+// **This variant of the FM sketch uses a set to store unique hashed values and a binary mask to track the maximum number of trailing zeroes.**
+// The estimated count of distinct values is calculated as 2^r * count, where 'r' is the maximum number of trailing zeroes observed and 'count' is the number of unique hashed values.
+// The underlying concept is that our hash function maps the input domain onto a logarithmic range. Each distinct value is mapped to 'i' with a probability of 2^-(i+1).
```

Comment:
1. In our context, I think it's not the hash function but the process of counting trailing zeros that has this effect.
2. I'm not sure if 2^-(i+1) can be called "logarithmic range".

Reply:
> 2\. I'm not sure if 2^-(i+1) can be called "logarithmic range".

I borrowed it from the paper.

Reply:
For the first one, I've added more details about how we achieve it to make it clearer.